src/Connect.js

```diff
@@ -149,16 +147,13 @@ class Connect {
           return Promise.resolve(Object.assign({id}, payload))
         }
         return this.verifyResponse(jwt).then(res => {
-          //TODO improve set/get state, address could be ethAddress
-          this.address = res.address
-          this.mnid = res.mnid
-          this.did = res.did
+          // Set identifiers present in the response
+          if (res.address) this.address = res.address
+          if (res.mnid) this.mnid = mnid
+          if (res.did) this.did = res.did
```

Comment:
did will always be returned, while if mind is returned, address will also always be returned